frontend/src/metabase/search/components/InfoText/InfoText.tsx

```diff
@@ -0,0 +1,186 @@
+import { t } from "ttag";
+import { useDatabaseQuery, useTableQuery } from "metabase/common/hooks";
+import {
+  browseDatabase,
+  browseSchema,
+  tableRowsQuery,
+} from "metabase/lib/urls";
+import Tooltip from "metabase/core/components/Tooltip";
+import { color } from "metabase/lib/colors";
+import { getRelativeTimeAbbreviated } from "metabase/lib/time";
+import { isNotNull } from "metabase/core/utils/types";
+import type { UserListResult } from "metabase-types/api";
+import { useUserListQuery } from "metabase/common/hooks/use-user-list-query";
+import { Icon } from "metabase/core/components/Icon";
+import { SearchResultLink } from "metabase/search/components/SearchResultLink";
+import type { WrappedResult } from "metabase/search/types";
+import { Group, Box, Text } from "metabase/ui";
+import type { InfoTextData } from "./get-info-text";
+import { getInfoText } from "./get-info-text";
+import {
+  DurationIcon,
+  LastEditedInfoText,
+  LastEditedInfoTooltip,
+} from "./InfoText.styled";
+
+export type InfoTextProps = {
+  result: WrappedResult;
+  isCompact?: boolean;
+};
+
+const LinkSeparator = (
+  <Box component="span" c="text.1">
+    <Icon name="chevronright" size={8} />
+  </Box>
+);
+
+export const InfoTextTableLink = ({ result }: InfoTextProps) => {
+  const { data: table } = useTableQuery({
+    id: result.table_id,
+  });
+
+  const link = tableRowsQuery(result.database_id, result.table_id);
+  const label = table?.display_name ?? null;
+
+  return (
+    <SearchResultLink key={label} href={link}>
+      {label}
+    </SearchResultLink>
+  );
+};
+
+export const InfoTextTablePath = ({ result }: InfoTextProps) => {
+  const { data: database, isLoading: isDatabaseLoading } = useDatabaseQuery({
+    id: result.database_id,
+  });
+
+  const showDatabaseLink =
+    !isDatabaseLoading && database && database.name !== null;
+  const showTableLink = showDatabaseLink && !!result.table_schema;
+
+  const getDatabaseLink = () => {
```

Comment:
Similarly to my other comment about `getDisplayComponent` in `RecentsList` - this function also is a component and should be defined as such.